### PR TITLE
Active Directory Query v2 Enhancements

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/37782

## Description
- Supports TLS secure connection
- Added *group* to the *member-type* argument in the ***ad-get-group-members*** command.
- Added the *sAMAccountName* argument to ***ad-get-group-members*** command.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 1]
```diff
@@ -684,16 +724,28 @@ def search_group_members(default_base_dn, page_size):
...
 
     custom_attributes: List[str] = []
-    default_attributes = DEFAULT_PERSON_ATTRIBUTES if member_type == 'person' else DEFAULT_COMPUTER_ATTRIBUTES
+
+    if member_type == 'person':
+        default_attributes = DEFAULT_PERSON_ATTRIBUTES
+    elif member_type == 'computer':
+        default_attributes = DEFAULT_COMPUTER_ATTRIBUTES
+    else:
+        default_attributes = DEFAULT_GROUP_ATTRIBUTES
```
Comment (yaakovi) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:736]: keeping the default as in the original code (should it be something else?).
```suggestion
    default_attribute_mapping = {
        'person': DEFAULT_PERSON_ATTRIBUTES,
        'group': DEFAULT_GROUP_ATTRIBUTES,
        'computer': DEFAULT_COMPUTER_ATTRIBUTES,
    }
    default_attributes = default_attribute_mapping.get(member_type, DEFAULT_COMPUTER_ATTRIBUTES)
```

**Review by yaakovi — COMMENTED** [review]
looks good, please add unit tests to the group type and if possible to the new connection flow

File: Packs/Active_Directory_Query/pack_metadata.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
     "name": "Active Directory Query",
     "description": "Active Directory Query integration enables you to access and manage Active Directory objects (users, contacts, and computers).",
     "support": "xsoar",
-    "currentVersion": "1.2.1",
+    "currentVersion": "1.2.2",
```
[the remark this answers is not in the document]
Comment (adi88d) [reply Packs/Active_Directory_Query/pack_metadata.json:5]: done

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 3]
```diff
@@ -1514,6 +1570,10 @@ def main():
             # here username should be the user dn
             conn = Connection(server, user=USERNAME, password=PASSWORD)
 
+        if SECURE_CONNECTION == 'TLS':
+            conn.open()
```
[the remark this answers is not in the document]
Comment (adi88d) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:1574]: the connection object doesn't have close method, they using `unbind`:
https://ldap3.readthedocs.io/en/latest/connection.html

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 4]
```diff
@@ -165,12 +165,12 @@ def endpoint_entry(computer_object, custom_attributes):
 
 
 def group_entry(group_object, custom_attributes):
-    # create an endpoint entry from a group object
+    # create an group entry from a group object
```
Comment (yaakovi) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:168]:
```suggestion
    # create a group entry from a group object
```

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/connection_test.py [hunk 5]
```diff
@@ -292,3 +292,18 @@ def add(self, *args, **kwargs):
     assert outputs.get('success') is False
     assert outputs.get('email') == 'test@paloaltonetworks.com'
     assert 'User must have a sAMAccountName' in outputs.get('errorMessage')
+
+
+def test_group_entry():
```
Comment (yaakovi) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/connection_test.py:297]: add one more test from the `search_group_members` function using the group entry.
also, use specific custom attributes.

**Review by yaakovi — APPROVED** [review]

File: Packs/Active_Directory_Query/ReleaseNotes/1_3_0.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+##### Active Directory Query v2
+- Supports TLS secure connection
```
Comment (roysagi) [inline Packs/Active_Directory_Query/ReleaseNotes/1_3_0.md:4]:
```suggestion
- Added support for TLS secure connections.
```

**Pull request merged**